src/oasim/rlwn_mod.f90

```diff
@@ -81,8 +82,11 @@ subroutine rlwn(km, lam, cosz, aw, bw, ac, bc, bpic, WtoQ, Ed, Es, h, phyto, exc
 call edeu(km, lam, aw, bw, ac, bc, bpic, WtoQ, Ed, Es, H, p, excdom, exdet, rmud, tirrq, cdomabsq, &
           avgq, sfceu)
 
-rn2 = rn*rn      !index refr squared  
-rlwnref = (1.0-rho)*sfceu/(rn2*Q)
+do i = 1,size(l_chan)
+   ind = MINLOC(abs(lam-l_chan(i)), DIM=1)
+   rn2 = rn*rn      !index refr squared  
```

Comment:
Does this need to move inside the loop? NOt sure if the `rn` values depend on `i` here

Reply:
you are right @kbhargava ! I just fixed it! Thanks